scripts/jenkins/qa_devstack.sh

```diff
@@ -157,4 +157,11 @@ tox -e smoke
 EOF
 fi
 
+if [ -z "${DO_UNSTACK}" ]; then
```

Comment:
either implement @jdsn suggestion or switch to -n  $DO_UNSTACK (-z is negating). otherwise this code doesn#t do anything in the way you implemented it (it is always false then)